### ISSUE TITLE
im.mak preventing build

### ISSUE THREAD

**Issue by viraladmin** [wavesplatform/matcher#450]
I am trying to compile this locally but I have been stuck for days. 

I have searched this code forewards, backwards, and inside out and cannot figure this out. 

I have all the dependencies compiled in my local maven repository. I have enabled resolvers += Resolver.mavenLocal in all the build files.

All goes fine... up until I hit the im.mak package. (im.mak.waves.transactions) in (dex/src/main/scala/com/wavesplatform/dex/tool/Checker.scala) 

I cannot get it to use my local build... I cannot find any references to this being called from any repos within the code. I see no references to it in my ~/.cache, ~/.m2/ nor ~/.ivy2/ folders.

The issue I keep having is that the package fails to compile saying it can't find package im.mak, which it obviously can't seeing as its not built in any of the sources nor linked from any of the dependency repos 

Yet it compiles perfectly when I don't try to do a local build... which baffles me. So how is this being included? 

More importantly how can I use a local build of the transactions-java and crypto-java packages? (I actually think its just the transactions-java. I believe the crypto-java is used to build the transactions-java but I don't think its specifically used by matcher source). 

Please help! This one has be stumped and my brain is going crazy searching this code trying to figure it all out. lol :)

**Comment by vsuharnikov** [comment]
>  I cannot find any references to this being called from any repos within the code.

Actually, this is a part of `waves-transactions`: https://search.maven.org/artifact/im.mak/waves-transactions/1.0.2/jar 
Sometimes dependencies fail to download because of availability issues of maven. 

> Yet it compiles perfectly when I don't try to do a local build

How do you do a local build? Also, could you show the log with the error?

> More importantly how can I use a local build of the transactions-java and crypto-java packages?

Did you mean https://github.com/wavesplatform/WavesJ ?

**Comment by cryptofunallday** [comment]
Unfortunately another perfect example of the documentation issue.  What he is experiencing is due to library incompatibilities which I got tired of trying to correct after realizing the documentation leads you down the wrong path.  Hacking this software together should not be.

**Comment by viraladmin** [comment]
Oh I got it sorted, and yes it was exactly an issue of me adding the wrong wavesj to my local repo..... I had to use the build "with dependencies".

@cryptofunallday I have understood the docs perfectly at every angle of compiling this. I am just doing things outside the scope of what has been intended. I have custom compiled the wallet,the etl (postgres), leveldb-jna, wavesj. waves-explorer, the waves data service, crypto java and transactions java, the protobuf schemes, match and the blockchain itself. 

My intent WAS to hack the software. 

We can consider this issue closed - but if anyone else has this issue, use the wavesj build that has all dependencies included.
